apis/python/src/tiledbsc/annotation_matrix.py

```diff
@@ -131,8 +131,16 @@ def _numpy_ndarray_or_scipy_sparse_csr_matrix(
             self._create_empty_array([matrix.dtype] * nattr, attr_names)
 
         df = pd.DataFrame(matrix, columns=attr_names)
-        with tiledb.open(self.uri, mode="w", ctx=self._ctx) as A:
-            A[dim_values] = df.to_dict(orient="list")
+        exc = None
```

Comment:
maybe https://github.com/litl/backoff would be better -- ?

Reply:
Another (more popular?) option is [tenacity](https://github.com/jd/tenacity), I had used it in the past as well its precursor ([retrying](https://github.com/rholder/retrying)).

Using any of these libraries is definitely better than copy-pasting the same boilerplate 15 times.